### PR TITLE
Support ad blocking

### LINKED ISSUE #7: Ad blocking
Would be awesome to have some kind of ad blocking to reduce noise in screenshots.

The best solution would be to use an existing ad blocker Chrome extension, like [uBlock Origin](https://github.com/gorhill/uBlock), but it seems it's only possible to load extension when not in headless mode (`{headless: false}`), so the Chrome window would show for a few seconds.

https://gist.github.com/sindresorhus/bca2f7d0c8b31205fa3c9f328d548c70
https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#working-with-chrome-extensions

We could potentially make it opt-in and let the user know the downside.

### PR DESCRIPTION
**Pull request by jopemachine**
Fixes #7.

 I think we could use `@cliqz/adblocker-puppeteer` package to support ad blocking.

# Notes

- This draft PR does not include the option representing whether to use the ad-blocker, related option, type, tests and documentations yet. 

- If I'm on the right way, I'd appreciate it if you could let me know. Then I will try to do the next works.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Yes, this looks like the right path.

What do you think about making ad-blocking opt-out instead of opt-in?

**Comment by jopemachine** [comment]
> Yes, this looks like the right path.
> 
> What do you think about making ad-blocking opt-out instead of opt-in?

In my opinion, opt-out would be better. I think there are likely to be more users who do not need ads in their screenshots than those who do not.

**Comment by sindresorhus** [comment]
Opt-out it is then.

File: index.d.ts [hunk 1]
```diff
@@ -219,6 +219,13 @@ export interface Options {
 	*/
 	readonly disableAnimations?: boolean;
 
+	/**
+	Disable [Ad Blocker](https://github.com/ghostery/adblocker/blob/master/packages/adblocker-puppeteer/README.md).
+
+	@default false
+	*/
+	readonly disableAdBlocker?: boolean;
```
Comment (sindresorhus) [inline index.d.ts:227]: I don't like double-negatives. Better to do `blockAds` and have it on by default.

`disableAnimations` is different as it describes the action, not the option itself.

**Comment by jopemachine** [comment]
@sindresorhus I've been thinking about the unit testing of this flag.
Do I need to compare generated screenshot having an ad with the one without the ad? (It seems that using [resemble.js](https://github.com/rsmbl/Resemble.js) is one way to do this)
Would it be there some better methods to test `blockAds` option?

**Comment by sindresorhus** [comment]
I think manually verifying it works is enough. A unit test will fail at some point when ads or ad blocker changes.

**Comment by jopemachine** [comment]
> I think manually verifying it works is enough. A unit test will fail at some point when ads or ad blocker changes.

Ok, I just made this Draft PR ready. Please let me know if there are other things for me to do :)

**Comment by sindresorhus** [comment]
Doesn't look like you updated the TS types.

**Comment by sindresorhus** [comment]
Thanks :)

Would you be able to add a `--no-block-ads` flag to the CLI?

**Comment by jopemachine** [comment]
> Thanks :)
> 
> Would you be able to add a `--no-block-ads` flag to the CLI?

Sure, I'll make a PR to CLI module as soon as capture-website's next version is released :)

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/capture-website/releases/tag/v2.3.0

**Pull request merged**